### PR TITLE
Add a test for EtherEncap module.

### PR DESCRIPTION
**Pull request by justinemarie**
Still going...

### CODE REVIEW

**Comment by justinemarie** [comment]
Am now using a random number generator to assign code reviews.

File: bessctl/conf/testing/module_tests/etherencap.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    metadata = SetMetadata(attrs=[{"name": "ether_src", "size": 6, "value_bin": "deadbeef1234".decode("hex")},
+        {"name": "ether_dst", "size": 6, "value_bin": "1234deadbeef".decode("hex")},
+        {"name" : "ether_type", "size": 2, "value_bin": "0800".decode("hex")}])
+    indriver -> metadata -> encap2 -> outdriver
+
+    bess.resume_all()
+    mysocket.send(bytes(test_packet_in))
+    return_data = mysocket.recv(2048)
+    bess.pause_all()
+    assert(str(return_data) == str(test_packet_out))
```
Comment (ddiproietto) [inline bessctl/conf/testing/module_tests/etherencap.py]: Maybe `bytes` instead of `str`?

File: bessctl/conf/testing/module_tests/etherencap.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            scapy.UDP,
+            "172.12.0.3",
+            "127.12.0.4")),
+        bytes(gen_packet(
+            scapy.TCP,
+            "192.168.32.4",
+            "1.2.3.4"))]
+    src -> Rewrite(templates=rwtemp) -> metadata -> encap1 -> Sink()
+    bess.resume_all()
+    time.sleep(15)
```
Comment (ddiproietto) [inline bessctl/conf/testing/module_tests/etherencap.py]: The other crash tests last 5 sec

File: bessctl/conf/testing/module_tests/etherencap.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+## Can't run Crash or Output tests because EtherEncap requires metadata to be set first #
+
+def etherencap_crashtest():
+    encap1 = EtherEncap()
+    src = Source()
+    metadata = SetMetadata(attrs=[{"name": "ether_src", "size": 6, "value_bin": "deadbeef1234".decode("hex")},
```
Comment (ddiproietto) [inline bessctl/conf/testing/module_tests/etherencap.py]: `decode` fails for me with python3

File: bessctl/conf/testing/module_tests/etherencap.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    sockname = "metadata_fixedvalues" + SCRIPT_STARTTIME
+    socket_port, mysocket = gen_socket_and_port(sockname)
+
+    indriver = PortInc(port=sockname)
+    outdriver = PortOut(port=sockname)
+
+    encap2 = EtherEncap()
+    metadata = SetMetadata(attrs=[{"name": "ether_src", "size": 6, "value_bin": binascii.unhexlify("deadbeef1234")},
+        {"name": "ether_dst", "size": 6, "value_bin": binascii.unhexlify("1234deadbeef")},
+        {"name" : "ether_type", "size": 2, "value_bin": binascii.unhexlify("0800")}])
```
Comment (sangjinhan) [inline bessctl/conf/testing/module_tests/etherencap.py]: Perhaps you can use `"value_int": 0x0800` to test it.

File: bessctl/conf/testing/module_tests/etherencap.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    sockname = "metadata_fixedvalues" + SCRIPT_STARTTIME
+    socket_port, mysocket = gen_socket_and_port(sockname)
+
+    indriver = PortInc(port=sockname)
+    outdriver = PortOut(port=sockname)
+
+    encap2 = EtherEncap()
+    metadata = SetMetadata(attrs=[{"name": "ether_src", "size": 6, "value_bin": binascii.unhexlify("deadbeef1234")},
+        {"name": "ether_dst", "size": 6, "value_bin": binascii.unhexlify("1234deadbeef")},
```
Comment (sangjinhan) [inline bessctl/conf/testing/module_tests/etherencap.py]: Does `bytearray.from('1234deadbeef')` work?

Comment (sangjinhan) [inline bessctl/conf/testing/module_tests/etherencap.py]: `fromhex`

**Comment by justinemarie** [comment]
fromhex doesn't work in python3; bytearray.from produces a bytearray not
bytes.

**Comment by sangjinhan** [comment]
`bytearray.fromhex()` should work both in 2 and 3. What I wonder is whether protobuf accept not only `str` and `bytes`, but also `bytearray`...

**Comment by justinemarie** [comment]
protobuf won't take bytearray, just bytes -- but could do some sort of
bytearray -> bytes

**Comment by justinemarie** [comment]
Okay, should be ready post-Travis.

**Comment by justinemarie** [comment]
*bump*

**Comment by ddiproietto** [comment]
LGTM thanks!

**Comment by ddiproietto** [comment]
gatto

**Pull request merged**